Mathlib/Analysis/Fourier/DirichletKernel.lean

```diff
@@ -0,0 +1,56 @@
+/-
+Copyright (c) 2025 Yi Yuan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yi Yuan
+-/
+module
+
+public import Mathlib.Analysis.Complex.Trigonometric
+
+/-!
+# Dirichlet kernel
+
+This file contains the definitions of Dirichlet kernel.
+
+## TODO : $S_n(f)(x) = (D_n * f)(x)$
+
+-/
+
+@[expose] public section
+
+open Real
+
+section
+
+theorem sum_cos_arith_progression (n : ℕ) (x a : ℝ) (hx : sin (x / 2) ≠ 0) :
```

Comment:
I think that by the [naming conventions](https://leanprover-community.github.io/contribute/naming.html), a more conventional name would be something like `sum_range_cos_mul_add` or something like that (and similarly below).